### DISCUSSION TITLE
why redaction removed text not belongs to the rect

### DISCUSSION THREAD

**Discussion by xulifan** [pymupdf/PyMuPDF#4029] [Looking for help]
Here is a simple code I used to show bottom part of a page in a new page:
```
def redaction_test(pdf_file_path, page_num, output_file_path):
    src_document = fitz.open(pdf_file_path)
    src_page = src_document[page_num]
    width = src_page.rect.width
    height = src_page.rect.height

    redaction_rect = src_page.rect
    redaction_rect.y1 = 300
    src_page.add_redact_annot(redaction_rect)
    src_page.apply_redactions()
    
    target_document = fitz.open()  # target doc
    target_page = target_document.new_page(width=width, height=height)
    target_page.draw_rect(redaction_rect, color=(1, 1, 0), fill=(1,1,0), width=2)
    
    show_rect = src_page.rect
    show_rect.y0 = redaction_rect.y1
    target_page.show_pdf_page(show_rect, src_document, page_num, clip=show_rect)
    target_document.save(output_file_path)
    src_document.close()
    target_document.close()
```

In the following screenshot, I was trying to add redaction to the top section of the left pdf file. The redaction rect is marked and filled with yellow shown in the right pdf document. However, some texts not belonging to the redaction rect becomes invisible. Although I can still select them and they are marked in light blue.
Anybody knows why?
The black stripes were added to cover sensitive information, not relevant to the questions.
![Screenshot 2024-11-06 at 11 52 26 AM](https://github.com/user-attachments/assets/3132ef50-8d36-4126-b10c-88569a262f16)

**Comment by jamie-lemon** [comment]
I believe this is a known issue - please see: https://github.com/pymupdf/PyMuPDF/issues/3751 , I tagged this discussion on there for reference. :)

**Reply by xulifan** [reply]
I can confirm with PyMuPDF 1.23.9, this issue does not exist.

**Comment by JorjMcKie** [comment]
Impossible to tell for sure without the file.
But we have a few cases with similar phenomena. For those, fixes are underway and will published with the next version.
So you may want to let me have the file via direct e-mail so I can check whether this will be fixed as part of that coming version.

**Reply by JorjMcKie** [reply]
Thanks for the file @xulifan - I tried it out: it should work in v1.25.0.

**Reply by xulifan** [reply]
Thanks, when will 1.25 be released?

**Reply by JorjMcKie** [reply]
thanks for the file, just tried it out for windows `('1.24.13', '1.24.10', '20241029000001')` and for Linux `('1.24.11', '1.24.10', '20241003000001')`.
It worked in both cases.
The critical thing is the MuPDF version (1.24.10 in both cases), because MuPDF does the actual redacting.

**Reply by xulifan** [reply]
Hi, I tried using the latest 1.24.14 on Mac, the problem still exists. Do you have any estimated timeline on the release of 1.25.0 to fix this issue. If it is not going to release soon, then I need to do something else for my current project.
